mozilla/blurts-server
# Fix #2210 - Update UI Tooling

This PR completes the following: 

- [x] Add/Copy Protocol Library via NPM
- [x] Migrate all CSS to SCSS, using gulp to compile
- [x] Update gitignore to ignore compiled CSS
- [x] Update stylelint for SCSS
- [x] Add `.dist` build step (as its used in prod) 

## Testing

- Remove your `/node_modules` folder and reinstall all NPM packages. 
- Run `npm start`
- **Expected:** 
  - **Visual:** No visible changes to the CSS/layout any/all pages.
  - **File Structure:** The CSS folder should be empty/reset to the following: 
  
```
.
+-- public/  
|   +-- css/ (contents are gitignored) 
|       app.css (compiled from app.scss located in `/public/scss/app.scss`)
|       +-- legacy/ (these files are not compiled and are used by email testing)
|           ***.css
|   +-- dist / (all contents are gitignored)
|       app.min.css (minified from compiled app.css)
```

See new documentation: https://github.com/mozilla/blurts-server/blob/2210-ui-tool-updates/docs/coding-standards.md
